test/colony-network-staking.js

```diff
@@ -29,20 +37,119 @@ contract("ColonyNetworkStaking", accounts => {
     await colonyNetwork.startNextCycle();
   });
 
-  async function accommodateChallengeAndInvalidateHash(test, round, idx, respondToChallenge = true) {
-    // Have our opponent respond to the challenge asked for
+  beforeEach(async () => {
+    goodClient = new ReputationMiningClient(MAIN_ACCOUNT, realProviderPort);
+    badClient = new MaliciousReputationMiningClient(OTHER_ACCOUNT, realProviderPort, 1, 0xfffffffff); // Mess up the second calculation. There will always be one if giveUserCLNYTokens has been called.
+    badClient2 = new MaliciousReputationMiningClient(accounts[2], realProviderPort, 1, 0xeeeeeeeee); // Mess up the second calculation in a different way
+    await goodClient.initialise(colonyNetwork.address);
+    await badClient.initialise(colonyNetwork.address);
+    await badClient2.initialise(colonyNetwork.address);
+
+    // Kick off reputation mining.
+    // TODO: Tests for the first reputation cycle (when log empty) should be done in another file
+    await giveUserCLNYTokens(colonyNetwork, MAIN_ACCOUNT, new BN("1000000000000000000"));
+
+    await clny.approve(colonyNetwork.address, "1000000000000000000");
+    await colonyNetwork.deposit("1000000000000000000");
+
+    const addr = await colonyNetwork.getReputationMiningCycle.call();
+    const repCycle = ReputationMiningCycle.at(addr);
+    await forwardTime(3600, this);
+    await repCycle.submitNewHash("0x0", 0, 10);
+    await repCycle.confirmNewHash(0);
+    // The inactive reputation log now has the reward for this miner, and the accepted state is empty.
+    // This is the same starting point for all tests.
+    const stakedBalance = await colonyNetwork.getStakedBalance.call(MAIN_ACCOUNT);
+    await colonyNetwork.withdraw(stakedBalance.toNumber());
+    const userBalance = await clny.balanceOf.call(MAIN_ACCOUNT);
+    await clny.transfer(0x0, userBalance, { from: MAIN_ACCOUNT });
```

Comment:
Why do we throw away these tokens?

Reply:
So that `MAIN_ACCOUNT` has no tokens, with nothing staked, like all other accounts, at the start of each test.